frame_utils.go

```diff
@@ -0,0 +1,75 @@
+package tchannel
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+	"sync"
+)
+
+type RecordingFramePool struct {
```

Comment:
nit: since this is now exported, suggest naming this more accurately, what do you think about `CheckedFramePool`?